apps/site/assets/ts/schedule/components/direction/GreenLineMenu.tsx

```diff
@@ -17,6 +20,32 @@ import iconGreen from "../../../../static/images/icon-green-line-small.svg";
 import { handleReactEnterKeyPress } from "../../../helpers/keyboard-events";
 import { getIsGlxOpen } from "../../../components/GlxOpen";
 
+type fetchAction =
+  | { type: "FETCH_COMPLETE"; payload: Route }
+  | { type: "FETCH_ERROR" }
+  | { type: "FETCH_STARTED" };
+
+export const fetchData = (
+  routeId: string,
+  dispatch: (action: fetchAction) => void
+): Promise<void> => {
+  dispatch({ type: "FETCH_STARTED" });
+  return (
+    window.fetch &&
+    window
+      .fetch(
+        `/schedules/map_api?id=${routeId}&direction_id=0`
```

Comment:
One option would be to set up a new endpoint custom made for this admittedly peculiar scenario. Like what if `/schedules/green_termini_api` could be called here and could return the destination values? This might be more straightforward than iterating through these calls and picking the data you need from those results.